scripts/check-selectors.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+# Set -u to error out if we use an unset variable.
+# Set -o pipefail to propagate errors in a pipeline.
+# Set -e to exit on any non-zero exit code.
+set -uo pipefail
+
+# Collection of repositories to check.
+declare -A repositories
+
+# is_passing is a boolean variable that is set to true if the check passes.
+is_passing=true
+
+# Define colors.
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# External dependencies: repositories, is_passing.
```

Comment:
nit: I would update this comment to describe in general terms the strategy that this function follows. You don't have to go in-depth but only provide a comment that someone can read and immediately get a sense of what the function accomplishes. Smth along these lines
```suggestion
#  check will goes through the list of repositories and for each repository it will extract metrics that exposes.
#  then in KM repo it will try to find expressions that contain those metrics and validate that they have 
#  the appropriate selectors.
```